.github/workflows/gradle.yaml

```diff
@@ -93,7 +93,7 @@ jobs:
       - name: Cache cargo dependencies
         uses: Swatinem/rust-cache@v2
         with:
-          shared-key: "tauri"
+          shared-key: "ubuntu-20.04"
```

Comment:
Pretty sure the cache already checks for OS

Reply:
https://github.com/Swatinem/rust-cache#cache-details

Reply:
> the github [job_id](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id),

It also does it per job_id, caches aren’t shared between different actions

Reply:
I don't think its a good idea to share the cache between different actions - to me it seems like its just asking for cache thrashing. Also, there will be a race condition on which cache gets uploaded, since the jobs run in parallel

Reply:
It's just sharing cache so both don't have to maintain different caches, they are not different systems (promised by Github).
Cache locking is something that already exists, only one can upload on it if both actions are being ran concurrently, although I'm sure the build GUI one will finish before the bundling one starts. The main idea is to save time and space.

Reply:
so if I understand you correctly:
* Github jobs run on the same machine, always. This means we are not sharing a cache between different runners that could thrash?
* These jobs share the same cache now, so we should see improvements

I didn't think github jobs worked that way - I thought jobs will run on any runner if its available, so they aren't necessarily running on the same machine.

If they did run on the same machine, you're right the lockfile in cargo will prevent concurrent modification. I wasn't really talking about that though -I was saying that one job will overwrite the cache of the other (thats if they *dont* run on the same machine though, if they run on the same the lockfile makes it not concurrent).

I think the only optimization I really see here is the space optimization not time. Does that make sense, or am I overlooking something?

If the only benefit is space optimization, I'm still not a big fan because I'm wary of cache thrashing (for example, running clippy before cargo build [thrashes cache](https://www.reillywood.com/blog/rust-faster-ci/). But maybe its not an issue?

Reply:
I'm gonna stamp it and we can deal with it later